src/covid_model_seiir_pipeline/workflow_template.py

```diff
@@ -2,50 +2,128 @@
+                 m_mem_free: str = None,
+                 num_cores: int = None,
+                 **kwargs):
+        if max_runtime_seconds is None:
+            max_runtime_seconds = self.default_runtime
+        if m_mem_free is None:
+            m_mem_free = self.default_memory
+        if num_cores is None:
+            num_cores = self.default_cores
+        super().__init__(num_cores=num_cores,
+                         queue=queue,
+                         max_runtime_seconds=max_runtime_seconds,
+                         m_mem_free=m_mem_free,
+                         **kwargs)
+
+    @abc.abstractmethod
+    @property
+    def default_runtime(self) -> int:
+        ...
+
+    @abc.abstractmethod
+    @property
+    def default_memory(self) -> str:
+        ...
+
+    @abc.abstractmethod
+    @property
+    def default_cores(self) -> int:
+        ...
+
+    def to_dict(self):
+        return self.to_wire()
+
+
+class WorkflowSpecification:
+
+    def __init__(self, project: str, task_specifications: Dict[str, ExecutorParameters]):
+        self.project = project
+        self.task_specifications = task_specifications
+
+    def to_dict(self):
+        return {
+            'project': self.project,
+            'task_specifications': {
+                task: spec.to_dict() for task, spec in self.task_specifications.items(),
+            }
+        }
+
 
 class TaskTemplate:
+    """Abstract factory for producing SEIIR pipeline tasks."""
 
-    task_name_template: str = None
-    command_template: str = None
-    params: ExecutorParameters = None
+    def __init__(self,
+                 task_name_template: str,
+                 command_template: str,
+                 params: ExecutorParameters,
+                 max_attempts: int = 1):
+        self.task_name_template = task_name_template
+        self.command_template = command_template
+        self.params = params
+        self.max_attempts = max_attempts
 
     def get_task(self, *_, **kwargs) -> BashTask:
         task = BashTask(
             command=self.command_template.format(**kwargs),
             name=self.task_name_template.format(**kwargs),
             executor_parameters=self.params,
-            max_attempts=1
+            max_attempts=self.max_attempts
         )
         return task
 
 
 class WorkflowTemplate:
 
-    workflow_name_template: str = None
-    task_templates: Dict[str, TaskTemplate] = None
-
-    def __init__(self, version: str):
+    def __init__(self, version: str, workflow_spec: WorkflowSpecification):
         self.version = version
+
         stdout, stderr = utilities.make_log_dirs(version, prefix='jobmon')
 
         self.workflow = Workflow(
             workflow_args=self.workflow_name_template.format(version=version),
-            project="proj_covid",
+            project=workflow_spec.project,
             stderr=stderr,
             stdout=stdout,
             seconds_until_timeout=60*60*24,
```

Comment:
Do you foresee any use in setting this to something shorter? My understanding of your covid workflow is a day is **way** too long for things to sit and you'd need to e.g., find another queue if it was taking this long